source/opt/ir_context.h

```diff
@@ -121,8 +133,61 @@ class IRContext {
   // Appends a function to this module.
   inline void AddFunction(std::unique_ptr<Function>&& f);
 
+  // Returns a pointer to a def-use manager.  If the def-use manager is
+  // invalid, it is rebuilt first.
+  opt::analysis::DefUseManager* get_def_use_mgr() {
+    if (!AreAnalysesValid(kAnalysisDefUse)) BuildDefUseManager();
+    return def_use_mgr_.get();
+  }
+
+  // Builds the def-use manager from scratch, even if it was already valid.
+  void BuildDefUseManager() {
+    def_use_mgr_.reset(new opt::analysis::DefUseManager(module()));
+    valid_analyses_ = (Analysis)(valid_analyses_ | kAnalysisDefUse);
```

Comment:
Prefer static_cast over c-style cast.

The masking and casting is ok, but we could also make an operator|  overload for this purpose.

Reply:
Defining the operator is a good idea.  I did that.